Add feedback popups

juliangiebel opened the pull request:
> [!NOTE]  
> This PR was reopened from the original PR here: #39415 
> The author requested their PR to be closed with the option for someone else to reopen the PR from their branch.

## About the PR
<!-- What did you change? -->
Added feedback popups that appear at the end of the round. Right now, these just link to the forms directing people where they should give feedback!

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
#38811 showed that a lot of people do not know about the forms and its very difficult for the average player to give feedback. This PR does a couple things:
1.) Makes it clear what PRs are still being tested out and are not finalized
2.) Make it very easy for any player to know where to give their feedback.

## Technical details
<!-- Summary of code changes for easier review. -->
System is pretty simple but my main issue was naming stuff. Sever sends messages to clients at the end of the round, clients listen for the messages and then display popups showing what PRs need feedback.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

https://github.com/user-attachments/assets/b29cc177-f5f7-4cd4-bb23-6e0c3f7a4d29

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl: Beck, Julian
- add: Added feedback popups that players get at the end of the round! They will list all "test" PRs or PRs that people want feedback on. Right now they will link to a form thread

Simyon264 commented on Content.Shared/FeedbackSystem/FeedbackPopupPrototype.cs:
```diff
@@ -0,0 +1,52 @@
+    /// Will be listed near the "Open Link" button; rich text is not allowed.
+    /// </summary>
+    [DataField]
+    public string? ResponseType;
```
I would make this an enum that is able to be localized

sleepyyapril replied:
> I would make this an enum that is able to be localized

that or a LocId

Simyon264 replied:
I suggested an Enum that way loc typos can be avoided. No need to duplicate 
"feedback-manager-type-sussy" everywhere

Simyon264 replied:
The difference is that entities have a separate system for localisation, these prototypes don't. 
Use an LocId then.

SlamBamActionman replied:
I am not talking about the entity `Name` and `Description` properties of entities, I am talking about this prototype's `string Title` and `string Description` properties. Unless you intend for those to be LocId too?

This would require that every feedback pop-up includes an .ftl file which I think is poor design of the system.

sleepyyapril reviewed (changes requested):
general UI critique:
1. try not to leave any blank space on the bottom of the window
2. maybe either don't open the window with no entries or add a little label that says "There is currently no feedback entries available to you." or something similar?
3. mayhaps use a StripeBack on the footer background? 
4. a horizontal line between every feedback entry would be nice or just a background on the feedback entries (a PanelContainer with a PanelDark StyleClass, for example)
5. setting a tooltip on "open link" to the actual link would also be nice

juliangiebel commented:
> I did some further testing and it seems that `showRoundEnd: false` prototypes don't show up at all. This would be because `GetOriginFeedbackPrototypes` is only called on roundend.

They have to be shown/added to a player through the commands. I can change that behavior if you want but them not automatically shown is done so admins can upload a prototype and only show it to specific players.
You can show them to multiple or all players at once using toolshed piping

juliangiebel commented:
> I think the UI could use some minor touch-ups but otherwise it all seems good - though I am not sure if the `StyleNano.cs` changes are relevant anymore?
> <img alt="image" width="696" height="646" src="https://private-user-images.githubusercontent.com/83650252/514754320-f3237581-49ea-49b7-bd92-d232f9f5b88f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMyMDkyMjEsIm5iZiI6MTc2MzIwODkyMSwicGF0aCI6Ii84MzY1MDI1Mi81MTQ3NTQzMjAtZjMyMzc1ODEtNDllYS00OWI3LWJkOTItZDIzMmY5ZjViODhmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE1VDEyMTUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhMmRkZTRmNWI4ODcwNjAzMGM0ODdiNzRjOTJkYjA2ZDJiMzkwMWYxYTVlOWVkZDhkMTU5MGFkNjRmODE4MTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mNVzaMS1cvCcJ8VJZp1ku62IUMnu6GkXaDwJ1R82BZw">
> 
> Did a quick photoshop test; perhaps something like this?
> <img alt="image" width="696" height="646" src="https://private-user-images.githubusercontent.com/83650252/514754132-53609ece-647c-415e-a7ad-b5fdf5071b3e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMyMDkyMjEsIm5iZiI6MTc2MzIwODkyMSwicGF0aCI6Ii84MzY1MDI1Mi81MTQ3NTQxMzItNTM2MDllY2UtNjQ3Yy00MTVlLWE3YWQtYjVmZGY1MDcxYjNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE1VDEyMTUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1Y2Q2MjE1ZTY4M2ViZmJiZDMxMzQ2YmNmNzMwNmY5NmQyZjNhM2U4YzAwZWZhNzE3Y2VhN2MzZDUyMDUyNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0-5xqJxX2YejA_brIWL0NadqLGv81cy4wLDT32Io25w">

Wth that looks nothing like my ui. The stylenano removal completely butchered it

juliangiebel commented:
I fixed the styling. It looks like this now because I changed the colors to the closest palette color
<img width="574" height="547" alt="Screenshot 2026-01-20 005251" src="https://github.com/user-attachments/assets/469372b6-a080-45b8-a5c7-4ecdb8599248" />

SlamBamActionman commented on Content.Shared/IoC/SharedContentIoC.cs:
```diff
@@ -1,4 +1,5 @@
-﻿using Content.Shared.Humanoid.Markings;
+﻿using Content.Shared.FeedbackSystem;
```
Looks like a left-over edit?

SlamBamActionman commented on Content.Shared/FeedbackSystem/FeedbackPopupPrototype.cs:
```diff
@@ -0,0 +1,52 @@
+    /// Should this feedback be shown when the round ends.
+    /// </summary>
+    [DataField]
+    public bool ShowRoundEnd = true;
```
Oh that makes sense 👍 Thanks!

alexalexmax commented:
hooray
